### PR TITLE
Support more placement options in TooltipTypes.d.ts

### PR DESCRIPTION
**Pull request by colynfulcrum**
Support more place types that floating-ui supports. These actually already work but I need the type to be updated to reflect existing allowed types.

### CODE REVIEW

File: src/components/Tooltip/TooltipTypes.d.ts
Comment (gabrieljablonski) [inline src/components/Tooltip/TooltipTypes.d.ts:15]:
```suggestion
export type PlacesType =
  | 'top'
  | 'right'
  | 'bottom'
  | 'left'
  | 'top-start'
  | 'top-end'
  | 'right-start'
  | 'right-end'
  | 'bottom-start'
  | 'bottom-end'
  | 'left-start'
  | 'left-end'
```

**Review by gabrieljablonski — CHANGES_REQUESTED** [review]

File: src/components/Tooltip/TooltipTypes.d.ts
Comment (gabrieljablonski) [inline src/components/Tooltip/TooltipTypes.d.ts:15]: just to be consistent with `floating-ui`

```suggestion
export type PlacesType =
  | 'top'
  | 'top-start'
  | 'top-end'
  | 'right'
  | 'right-start'
  | 'right-end'
  | 'bottom'
  | 'bottom-start'
  | 'bottom-end'
  | 'left'
  | 'left-start'
  | 'left-end'
```

**Review by gabrieljablonski — CHANGES_REQUESTED** [review]

**Comment by gabrieljablonski** [comment]
Thanks for suggesting this. Before we can merge this, we have to update a few more files.

Source:
- `src/utils/compute-positions-types.d.ts`
Docs:
- `docs/options.mdx`
- `docs/examples/place.mdx`

I believe I haven't missed any. Feel free to update them, or leave it to us for whenever we get the time.

**Comment by colynfulcrum** [comment]
> Thanks for suggesting this. Before we can merge this, we have to update a few more files.
> 
> Source:
> 
> * `src/utils/compute-positions-types.d.ts`
> 
> Docs:
> 
> * `docs/options.mdx`
> * `docs/examples/place.mdx`
> 
> I believe I haven't missed any. Feel free to update them, or leave it to us for whenever we get the time.

I've updated those files. Hope that works. Have a great weekend!

**Comment by colynfulcrum** [comment]
@gabrieljablonski sorry, I missed a big section in the docs. I will get to this later tonight.

image

**Comment by colynfulcrum** [comment]
@gabrieljablonski I updated the examples to include the other types... see screen recording

https://github.com/ReactTooltip/react-tooltip/assets/122113864/93c0de89-ed45-49de-a518-1d3e03775bc9

**Review by gabrieljablonski — APPROVED** [review]

**Review by danielbarion — APPROVED** [review]

**Comment by danielbarion** [comment]
Available at `v5.15.1`

**Pull request merged**
